x/params/simulation/msgs.go

```diff
@@ -0,0 +1,123 @@
+package simulation
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+)
+
+type simParamChange struct {
+	subspace string
+	key      string
+	subkey   string
+	simValue func(r *rand.Rand) string
+}
+
+// paramChangePool defines a static slice of possible simulated parameter changes
+// where each simParamChange corresponds to a ParamChange with a simValue
+// function to generate a simulated new value.
+//
+// NOTE: All parameter value ranges are adapted from appStateRandomizedFn.
+//
+// TODO: governance parameters (blocked on an upgrade to go-amino)
+var paramChangePool = []simParamChange{
```

Comment:
are these the only parameters which can be changed?

maybe we should only allow parameter changes that we've simulated

Reply:
These are not the whole kit and caboodle -- I've simulated a majority, but not all.

Reply:
hmm, ok - which haven't we simulated / might they be likely to happen in practice?

Reply:
Mostly `gov` -- I'm waiting on https://github.com/cosmos/cosmos-sdk/issues/4247 to be finished first before this.